src/cc65/symtab.c

```diff
@@ -867,9 +867,10 @@ SymEntry* AddBitField (const char* Name, const Type* T, unsigned Offs,
         Entry->V.B.BitOffs  = BitOffs;
         Entry->V.B.BitWidth = BitWidth;
 
-        if (!SignednessSpecified) {
+        if (!SignednessSpecified && !IsTypeChar (Entry->Type)) {
```

Comment:
This way of handling seems to be coupled with the underlying type of `char` (it could be `signed char` if compiled with `--signed-chars`).

I think whatever underlying signedness should be "normalized" to `unsigned` as long as it is not specified explicitly, for the reason that `char` bit-fields are integers rather than characters. This would keep consistency with other standard-integer-type bit-fields but leave `enum`-type bit-fields unique (for a sensible reason).

Reply:
> `char` bit-fields are integers rather than characters.

What do you mean by this? `char` is both a character type and an integral type.

Reply:
https://port70.net/~nsz/c/c99/n1256.html#6.7.2.1p9
> A bit-field is interpreted as a signed or unsigned integer type consisting of the specified number of bits.

I consider the idea "char bit-fields" to be an extension to allow (`signed`/`unsigned`) `char` to be used to declare bit-fields, which have no other difference from corresponding standard (`signed`/`unsigned`) `int` bit-fields other than how they are declared, which are just required to be interpreted as integer types but not character types.

Reply:
Anyway, I made `char` unsigned in bit-fields, even with `--signed-chars`.